File: hw1/Avg.java
import java.io.*;
import java.util.Scanner;

public class Avg {
    public static void main(String args[]) {
        String amtStr;
        int count = 0;
        int sum = 0;
        int grade = 0;
        double average = 0.0;

        Scanner sc = new Scanner(System.in);
        while(true) {
            System.out.println("Please input your score:");
            amtStr = sc.next();    // Read the input string
            
            int length = amtStr.length();   // Get the length of string
            
            try {    // Convert string to integer
                grade = Integer.parseInt(amtStr);
            }
            catch (NumberFormatException e) {   // Error: not integer
                System.out.println("Illegal input, ending the program...");
                System.exit(1);
            }

            if(grade == -1 && count != 0)   // Compute the average
                break;

            if(grade < 0 || grade > 100) {   // Out of boundary (0-100)
                System.out.println("Illegal input, ending the program...");
                System.exit(1);
            }
            sum += grade;
            count++;
        }

        // Compute the average
        average = (double) sum / count;
        System.out.printf("Your average score is: %.2f\n", average);
    }
}

File: hw1/Avg.py
count = 0
summation = 0
grade = 0
average = 0.0

while True:
    string = input('Please input your score:\n')  # Read the input string
    length = len(string)     # Get the length of string

    # Check the illegal input
    for i in range(length):
        if i == 0 and string[i] == '-':
            continue
        if not(string[i] >= '0' and string[i] <= '9'):
            print('Illegal input, ending the program...')
            exit()

    grade = int(string)   # Convert string to int

    if grade == -1 and count != 0:    # Compute the average
        break

    # Check the boundary (0-100)
    if grade < 0 or grade > 100:
        print('Illegal input, ending the program...')
        exit()

    summation += grade
    count += 1

# Compute the average
average =  summation / count
print('Your average score is: {:.2f}'.format(average))

